### PR TITLE
RNG seeds

### PR DESCRIPTION
**Pull request by MarcBerliner**
> [!IMPORTANT]
> Add `random.py` for time-based random seed generation and management, with corresponding tests and version reset in `pyproject.toml`.
> 
>   - **New Features**:
>     - Add `random.py` with functions `current_time_integer()`, `generate_seed()`, `seed()`, and `get_seed()` for handling random seeds using current time.
>   - **Tests**:
>     - Add `test_random.py` to test `current_time_integer()`, `generate_seed()`, and `seed()` functions.
>   - **Misc**:
>     - Import `random` module in `__init__.py`.
>     - Reset version to `0.0.0` in `pyproject.toml`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ionworks%2Fiwutil&utm_source=github&utm_medium=referral)<sup> for 2979ffb8d365362b34b6ef5be9037235e81f5f87. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ authors = [{ name = "Ionworks Technologies Inc", email = "info@ionworks.com" }]
 readme = { file = "README.md", content-type = "text/markdown" }
 license = { file = "LICENSE.md" }
 requires-python = ">=3.10"
-version = "0.3.4"
+version = "0.3.7"
```
Comment (kratman) [inline pyproject.toml:8]: This will avoid the errors again until I have time to switch it over to correct versioning in the main branch
```suggestion
version = "0.0.0"
```

File: tests/test_random.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    assert edge_seed2 < 2**32  # Valid seed range
+    # Verify our wraparound logic would accept both
+    mock_time1_mod = max_uint32
+    mock_time2_mod = 1
+    assert (
+        edge_seed1 >= mock_time1_mod or edge_seed1 <= mock_time2_mod
+    )  # Valid wraparound
+    assert (
+        edge_seed2 >= mock_time1_mod or edge_seed2 <= mock_time2_mod
+    )  # Valid wraparound
```
[the remark this answers is not in the document]
Comment (MarcBerliner) [reply tests/test_random.py:111]: Thanks, I was testing the edge cases and forgot to remove
